src/repository.py

```diff
@@ -34,6 +41,47 @@
 CONFIG_USER_EMAIL = (CONFIG_USER_SECTION_NAME, "email")
 
 
+class DiffSummary(NamedTuple):
+    is_dirty: bool
+    new: Sequence[str]
+    removed: Sequence[str]
+    modified: Sequence[str]
+
+    @classmethod
+    def from_raw_diff(cls, diffs: Sequence[Diff]):
+        new_files = {diff.a_path for diff in diffs if diff.new_file}
+        removed_files = {diff.a_path for diff in diffs if diff.deleted_file}
+        modified_files = {diff.a_path for diff in diffs if diff.renamed_file}.union(
+            diff.a_path for diff in diffs if diff.change_type == "M"
+        )
+
+        return DiffSummary(
+            is_dirty=len(diffs) > 0,
+            new=list(new_files),
```

Comment:
Should these be tuples rather than lists since they should be constant?

Reply:
Uhm, good point. But since we also don't want items to repeat, maybe frozensets then?

Reply:
yeh sounds good as well